Optimization of string splitting

shoonia opened the pull request:
Added a limit for split operation. It decreases the count of the new strings.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/13812"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

nicolo-ribaudo commented:
Could you share some microbenchmarks that show how much this improves the performance of `@babel/code-frame`?

nicolo-ribaudo commented on packages/babel-code-frame/src/index.ts:
```diff
@@ -147,7 +147,7 @@ export function codeFrameColumns(
   let frame = highlightedLines
-    .split(NEWLINE)
+    .split(NEWLINE, end)
     .slice(start, end)
```
Do we still need `, end` here?

shoonia replied:
It's limit for `.split()`. If we have, supposed, 100 lines and need to grab from 30 to 34. We know the end line it's a 34, so the `.split()` creates the first 34 lines and stops.

shoonia commented:
Simple benchmark: https://www.measurethat.net/Benchmarks/Show/15166/0/stringsplit-with-limit-3

```
split without limit:    x 2,376,932 ops/sec ±6.32% (57 runs sampled)
split with limit:       x 3,391,328 ops/sec ±0.67% (64 runs sampled)
```

nicolo-ribaudo commented:
Thanks!
